net/dcwapd/files/dcwapd.init

```diff
@@ -254,3 +249,69 @@ datachannel_down()
 		#`/etc/init.d/network restart`
 	fi
 }
+
+start_service() {
+	get_channelsets
+	# get the list of channel sets
+	channelsets=$result
+
+	for channelset in $channelsets; do
+	if [ -n "$channelset" ]; then
+		get_channelset_enabled "$channelset"
+		enabled=$result
+		if [ "$enabled" = "1" ]; then
+			# the channel set is enabled
+
+			# get the list of data channels used by the channel set
+			get_datachannels "$channelset"
+			datachannels=$result
+			for datachannel in $datachannels; do
+				datachannel_up "$datachannel"
+			done
+		fi
+	fi
+	done
+
+	procd_open_instance
+	procd_set_param file /etc/config/dcwapd
+	procd_set_param command dcwapd
```

Comment:
I am curious. Does this run in the foreground?

Reply:
I believe that it should now

Reply:
Excellent. daemons normally fork to the background. This interferes with procd which expects to have direct control.

Reply:
I think that I had it forking until that last commit, so thanks for verifying

Reply:
I'm pretty sure that the `dcwapd` daemon itself has always run in the foreground.